Same problem https://github.com/NixOS/nixpkgs/pull/28662

```
In file included from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/compiler.h:54:0,
                 from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/uapi/linux/stddef.h:1,
                 from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/stddef.h:4,
                 from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/uapi/linux/posix_types.h:4,
                 from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/uapi/linux/types.h:13,
                 from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/types.h:5,
                 from /tmp/nix-build-ixgbevf-4.2.1-3.18.0.drv-0/ixgbevf-4.2.1/src/ixgbevf_main.c:29:
/nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/compiler-gcc.h:107:30: fatal error: linux/compiler-gcc6.h: No such file or directory
 #include gcc_header(__GNUC__)
                              ^
compilation terminated.
```

--- a/pkgs/os-specific/linux/ixgbevf/default.nix
+++ b/pkgs/os-specific/linux/ixgbevf/default.nix
@@ -2,11 +2,11 @@
 
 stdenv.mkDerivation rec {
   name = "ixgbevf-${version}-${kernel.version}";
-  version = "4.1.2";
+  version = "4.2.1";
 
   src = fetchurl {
     url = "mirror://sourceforge/e1000/ixgbevf-${version}.tar.gz";
-    sha256 = "1dismhiq0asf04rv6pv2sk2m3xcy6m3bpk16gmxqybca3xa28a5b";
+    sha256 = "1666hsavaspjmf45ij396mkndzk6g0n3ibr2glfdhjfqhfy35zb8";
   };
 
   hardeningDisable = [ "pic" ];
@@ -24,5 +24,6 @@ stdenv.mkDerivation rec {
     homepage = https://sourceforge.net/projects/e1000/files/ixgbevf%20stable/;
     license = stdenv.lib.licenses.gpl2;
     priority = 20;
+    broken = (stdenv.lib.versionOlder kernel.version "4.9");
   };
 }

